File: recsdk_tf_2.0/examples/DLRM/model/logger.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# Copyright 2025. Huawei Technologies Co.,Ltd. All rights reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# ==============================================================================

import logging


def _get_logger(log_level: str) -> logging.Logger:
    dlrm_logger = logging.getLogger("dlrm_model")
    formatter = logging.Formatter(fmt="[%(asctime)s] [%(levelname)s] %(message)s", datefmt="%m/%d/%Y %H:%M:%S %p")
    stream_handler = logging.StreamHandler()
    stream_handler.setFormatter(formatter)
    dlrm_logger.addHandler(stream_handler)
    dlrm_logger.setLevel(log_level)
    return dlrm_logger


logger = _get_logger("DEBUG")

File: training/common/src/core/log/logger.cpp
/* Copyright 2024. Huawei Technologies Co.,Ltd. All rights reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
        limitations under the License.
==============================================================================*/


#include "logger.h"

namespace MxRec {

int MxRec::Logger::level = MxRec::Logger::INFO;
int MxRec::Logger::rank = 0;

void Logger::SetRank(int logRank)
{
    Logger::rank = logRank;
}

void Logger::SetLevel(int logLevel)
{
    Logger::level = logLevel;
}

int Logger::GetLevel()
{
    return Logger::level;
}

const char* Logger::LevelToStr(int logLevel)
{
    if (logLevel < TRACE || logLevel > ERROR) {
        return "INVALID LEVEL";
    }
    static const char* msg[] = {
        "TRACE",
        "DEBUG",
        "INFO",
        "WARN",
        "ERROR",
    };
    constexpr int levelOffset = 2;
    return msg[logLevel + levelOffset];
}

void Logger::LogUnpack(std::queue<std::string>& fmt, std::stringstream &ss)
{
    while (!fmt.empty()) {
        ss << fmt.front();
        fmt.pop();
    }
    return;
}

}